pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -312,13 +318,101 @@ func (o *UpgradeBootOptions) updateBootConfig(versionStreamURL string, versionSt
 		return errors.Wrapf(err, "failed to fetch upgrade tag %s from %s", upgradeVersion, bootConfigURL)
 	}
 
+	// Set up custom merge driver to ensure that specified files always use the local/dev env version in merges/cherry picks with conflicts
+	err = o.configureGitMergeExcludes()
+	if err != nil {
+		return errors.Wrapf(err, "configuring files (%s) to always use existing version when conflicting with upstream", strings.Join(filesExcludedFromCherryPick, ", "))
+	}
+
 	err = o.cherryPickCommits(configCloneDir, currentSha, upgradeSha)
 	if err != nil {
 		return errors.Wrap(err, "failed to cherry pick upgrade commits")
 	}
-	err = o.excludeFiles(currentSha)
+
+	err = o.removeMergeExcludesFromAttributes()
```

Comment:
this needs to happen in a defer, otherwise the cleanup is not happening if a cherry-pick error occurs.